fiona/_transform.pyx

```diff
@@ -263,3 +264,63 @@ def _transform_geom(src_crs, dst_crs, geom, antimeridian_cutting, antimeridian_o
     OSRRelease(dst)
 
     return out_geom
+
+
+from libcpp.unordered_map cimport unordered_map
+
+cdef OGRGeometryH segmentize(OGRGeometryH geom, double max_length):
+    cdef OGRGeometryH new_geom = NULL
+    new_geom = OGR_G_Clone(geom)
+    OGR_G_Segmentize(new_geom, max_length)
+    return new_geom
+
+
+ctypedef OGRGeometryH (*geometry_func)(OGRGeometryH, double)
+cdef unordered_map[int, geometry_func] func_map
+
+func_map[1] = <geometry_func>segmentize
+
+cdef geometry_func func_alias(int choice):
+    return func_map[choice]
+
+
+def apply_geom(operations, geoms):
+    """Apply a series of geometry transforming operations to one or more geometries.
+
+    Parameters
+    ----------
+    operations: list
+        A list of operation names and extra arg as a tuple.
+    geoms: Geometry or Sequence[Geometry]
+        Geometries on which to apply operations.
+
+    Yields
+    ------
+    Geometry
+
+    """
+    cdef OGRGeometryH ogr_geom1 = NULL
+    cdef OGRGeometryH ogr_geom2 = NULL
+    cdef int choice = 0
+
+    op_map = {
+        "segmentize": 1
+    }
+
+    for geom in geoms:
+        ogr_geom1 = _geometry.OGRGeomBuilder().build(geom)
+
+        for op_name, op_arg in operations:
```

Comment:
Basically it's a per-geometry pipeline implemented in C.

The other obvious design is to add segmentize &c as methods to our Geometry class. But performance won't be great until version 2.0 of fiona when Geometry is an extension class and carries an OGRGeometryH or a zero-copy data structure around with it.

Reply:
Might not be bad to wait until 2.0 for this as the `Geometry` class seems to have the potential to significantly impact on the design.